libtenzir/builtins/operators/chart2.cpp

```diff
@@ -370,14 +370,20 @@ class chart_operator2 final : public crtp_operator<chart_operator2> {
           }
           return *gnames.emplace(value_at(string_type{}, *gs.array, idx)).first;
         });
-        auto* newb = get_bucket(groups, plugins, x, group_name, s);
-        if (b != newb) {
+        auto [newb, new_bucket] = get_bucket(groups, x, group_name, s);
+        if (b != newb or new_bucket) {
           if (b) {
             for (auto&& instance : *b) {
               instance->update(subslice(slice, consumed, consumed + i), s);
             }
           }
-          b = newb;
+          if (new_bucket) {
+            b = &get_groups(groups, x, s)
+                   ->emplace(group_name, args_.make_bucket(plugins, s))
```

Comment:
Can `get_groups` return `nullptr` here? Looks like it.

Reply:
Nope, we only return true from `get_bucket` if the group exists but the bucket does not.